torch/fx/experimental/symbolic_shapes.py

```diff
@@ -318,12 +318,12 @@ def check_supported_type(x):
             # difficult to check the types.
             if base.is_zero:
                 return sympy.S.Zero
-            if divisor == 1:
-                return sympy.floor(base)
             if isinstance(base, sympy.Integer) and isinstance(divisor, sympy.Integer):
                 return base // divisor
             if isinstance(base, (sympy.Integer, sympy.Float)) and isinstance(divisor, (sympy.Integer, sympy.Float)):
                 return sympy.floor(base / divisor)
+            if divisor == 1:
+                return base
```

Comment:
@nkaretnikov The issue here is that if you have something like `x//1`, we end up with `floor(x)`, which is undesirable.

Reply:
@Chillee undesirable in what sense? do you have a repro or an example in mind?

Reply:
@nkaretnikov Well, for one, if it's an integer, it's just redundant :)

The other thing (and why I did this change in the first place) is that it's causing issues in Inductor when a shape has a `floor` expression in it.

Reply:
@Chillee Okay, no action is necessary for this PR since it's FloorDiv-specific, but could you please open an issue with a repro for the floor problem you mention?

I think I might have seen it too. I feel like FloorDiv should implement the Python spec as close as possible and we should just fix the floor issue. The problem I've seen is this: sym_floor couldn't handle exprs with Symbols in it (because math.floor doesn't), so I extended it similar to sym_sqrt. But it didn't work: floor appeared in cpp/triton codegen as just floor instead of, say, tl.something.floor. It might be that I did it incorrectly, though.

Reply:
Currently our CI doesn't test through Inductor with dynamic shapes (which is where it breaks).

Reply:
Good to know! I actually started working on that based on Ed's suggestion here: https://github.com/pytorch/pytorch/pull/91952#discussion_r1070724600 But it's still not done and I only have the code locally. I'll do more FloorDiv testing once that lands, then.